### DISCUSSION TITLE
Simplest Popup Example?

### DISCUSSION THREAD

**Discussion by plettb** [CommunityToolkit/Maui#391] [Q&A] [answered]
The "CommunityToolkit.Maui.Sample" project is great, and does what I need. However, I'm having a difficult time distilling _only_ what I need from it.  I can't get a simple popup, using "ShowPopupAsync" to display in a completely new project.  I don't understand what's happening.  I get zero feedback - no exception, no debug output, nothing.  The popup just doesn't become visible, and the "ShowPopupAsync" never returns.  Also, while the system appears to think the popup _is_ visible, I can still click the button that initiates the popup.  Any ideas?

Ideally what I'd like is the simplest possible project that demonstrates the use of the simplest possible modal popup.  Help would be appreciated.

**Comment by bijington** [comment]
Have you checked out our docs? There is a very simple example of how to use `ShowPopupAsync` here: https://docs.microsoft.com/en-gb/dotnet/communitytoolkit/maui/views/popup#returning-a-result

If you are still stuck are you able to share the code you are using to help us see if it is an issue?

**Reply by plettb** [reply]
Thanks a lot for this.  I was unaware that documentation already existed - I guess I should have looked further.  Egg on my face for needing to be told to RTFM! :)

It turns out this is actually some kind of weird interaction with the debugger.  If I just let the app run, I see the popup, but if I break between the button click and the popup showing, it never shows.  This sounds like some kind of visualization bug to me, but I'm guessing it's already known.

Also, when the app is in the state above, where the button has been pushed, the breakpoint has been encountered and continued, and the popup should be showing, the other controls are still visible and active.  So a user can actually push the button multiple times, hitting the breakpoint each time, but never getting out of the click event handler.

If this is not a known bug, please let me know and I'll post a trivial app with instructions on how to replicate the problem.

Thanks again!

P.S.  Now that I've figured out it's a weird interaction with the debugger, I find I can cause the same effect in the sample app.  Simply put a breakpoint in one of the click handlers in "MultiplePopupPage".

**Reply by pictos** [reply]
@plettb are you running on Windows?

**Reply by plettb** [reply]
Yes.

**Reply by pictos** [reply]
@plettb so on Windows we use the Flyout, and if you move to another screen the popup closes, it's this way by design.

**Comment by VladislavAntonyuk** [comment, marked as answer]
You can track it here: https://github.com/CommunityToolkit/Maui/issues/371

**Comment by CliffAgius** [comment]
Closed as answered
